src/Engine/Graphics/Outdoor.cpp

```diff
@@ -2371,79 +2371,70 @@ void ODM_ProcessPartyActions() {
                     pODMFace->pFacePlaneOLD.SignedDistanceTo(new_pos_low_x, new_pos_low_y, new_pos_low_z);
                 if (v55 > 0) {
-                    party_new_x = new_pos_low_x + fixpoint_mul(pODMFace->pFacePlaneOLD.vNormal.x, v55);
-                    party_new_Y = new_pos_low_y + fixpoint_mul(pODMFace->pFacePlaneOLD.vNormal.y, v55);
+                    partyNewX = new_pos_low_x + fixpoint_mul(pODMFace->pFacePlaneOLD.vNormal.x, v55);
+                    partyNewY = new_pos_low_y + fixpoint_mul(pODMFace->pFacePlaneOLD.vNormal.y, v55);
                     if (!bFaceSlopeTooSteep)
-                        party_new_Z = new_pos_low_z + fixpoint_mul(pODMFace->pFacePlaneOLD.vNormal.z, v55);
-                }
-                if (pParty->floor_face_pid != collision_state.pid &&
-                    pODMFace->Pressure_Plate()) {
-                    pParty->floor_face_pid = collision_state.pid;
-                    trigger_id = pODMFace->sCogTriggeredID;  //
+                        partyNewZ = new_pos_low_z + fixpoint_mul(pODMFace->pFacePlaneOLD.vNormal.z, v55);
                 }
             }
 
             if (pODMFace->uPolygonType == POLYGON_InBetweenFloorAndWall) {
-                int dot = abs(party_y_speed * pODMFace->pFacePlaneOLD.vNormal.y +
-                           party_z_speed * pODMFace->pFacePlaneOLD.vNormal.z +
-                           party_x_speed * pODMFace->pFacePlaneOLD.vNormal.x) >> 16;
+                pParty->bFlying = false;
+                pParty->uFlags &= ~PARTY_FLAGS_1_LANDING;
+
+                // this pushes party slightly up away from the surface so you can climb it
+                int dot = abs(partyInputYSpeed * pODMFace->pFacePlaneOLD.vNormal.y +
+                           partyInputZSpeed * pODMFace->pFacePlaneOLD.vNormal.z +
+                           partyInputXSpeed * pODMFace->pFacePlaneOLD.vNormal.x) >> 16;
                 if ((collision_state.speed / 8) > dot)
                     dot = collision_state.speed / 8;
-                party_x_speed += fixpoint_mul(dot, pODMFace->pFacePlaneOLD.vNormal.x);
-                party_y_speed += fixpoint_mul(dot, pODMFace->pFacePlaneOLD.vNormal.y);
-                party_z_speed += fixpoint_mul(dot, pODMFace->pFacePlaneOLD.vNormal.z);
-                bpartyslope = true;
-                if (party_x_speed * party_x_speed + party_y_speed * party_y_speed >= 400) {
-                    if (pParty->floor_face_pid != collision_state.pid &&
-                        pODMFace->Pressure_Plate()) {
-                        pParty->floor_face_pid = collision_state.pid;
-                        trigger_id = pODMFace->sCogTriggeredID;  //
-                    }
-                } else {
-                    party_x_speed = 0;
-                    party_z_speed = 0;
-                    party_y_speed = 0;
-                }
+                partyInputZSpeed += fixpoint_mul(dot, pODMFace->pFacePlaneOLD.vNormal.z);
```

Comment:
Why are we not changing x and y here?

Reply:
From my limited look into collisions its not strictly needed here but will re-add for time being